smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/HttpProtocolTestGenerator.java

```diff
@@ -333,10 +338,84 @@ private void generateServerRequestTest(OperationShape operation, HttpRequestTest
         });
     }
 
-    private ObjectNode buildQueryBag(HttpRequestTestCase testCase) {
+    private void generateMalformedRequestTest(OperationShape operation, HttpMalformedRequestTestCase testCase) {
+        Symbol operationSymbol = symbolProvider.toSymbol(operation);
+
+        Map<String, String> requestHeaders = testCase.getRequest().getHeaders().entrySet().stream()
+                .map(entry -> new Pair<>(entry.getKey().toLowerCase(Locale.US), entry.getValue()))
+                .collect(MapUtils.toUnmodifiableMap(Pair::getLeft, Pair::getRight));
+        String queryParameters = Node.prettyPrintJson(buildQueryBag(testCase.getRequest().getQueryParams()));
+        String requestHeaderParameters = Node.prettyPrintJson(ObjectNode.fromStringMap(requestHeaders));
+        String requestBody = testCase.getRequest().getBody().orElse(null);
+
+        String testName = testCase.getId() + ":MalformedRequest";
+        testCase.getDocumentation().ifPresent(writer::writeDocs);
+        writer.openBlock("it($S, async () => {", "});\n", testName, () -> {
```

Comment:
We'll probably want to add filtering ability here. Unfortunately it'll need to be a separate set of filters.

Reply:
Yeah, I wrote it locally yesterday.